### ISSUE TITLE
DelayedBase.__repr__ does not use repr?

### ISSUE THREAD

**Issue by albertz** [rwth-i6/sisyphus#104]
https://github.com/rwth-i6/sisyphus/blob/a5de7fe9e8584251c74be37c0a18c61102cc14cb/sisyphus/delayed_ops.py#L77
```python
class DelayedBase:
    ...

    def __repr__(self):
        return str(self.get())

    def __str__(self):
        return str(self.get())
```
Why is there `str(...)` for the `__repr__`? Isn't this inconsistent? Shouldn't it be `repr`?

**Comment by critias** [comment]
I agree, it looks like a copy&paste bug. Fixing it could cause to break some recipes 🤔 but I guess it should be done anyway.

**Comment by michelwi** [comment]
I think if we decided to break stuff, then we should put an effort to make the behavior of `DelayedBase` and other DelayedOps consistent to those of `Path` 
https://github.com/rwth-i6/sisyphus/blob/5e8468dcccd7b8800efa646311fb1b4111b1c755/sisyphus/job_path.py#L260-L264
and `Variable`
https://github.com/rwth-i6/sisyphus/blob/5e8468dcccd7b8800efa646311fb1b4111b1c755/sisyphus/job_path.py#L443-L450

i.e. the `__str__` function serializes the object to its usable form, but may only be used by a worker. the `__repr__` may also be used in the manager for logging or displaying on the gui, but should be unusable "by accident" to prevent hash-bugs.

Also we should be more rigorous in applying the `check_is_worker` function.

>  Fixing it could cause to break some recipes

It will definitively break things in i6_core, but a am willing to spend time to fix this. It has cost us a lot of time tracing down dubious bugs in the past.
